### ISSUE TITLE
VFolders: Stats must exclude obsolete units

### ISSUE THREAD

**Issue by unho** [translate/pootle#3736]
In https://github.com/translate/pootle/blob/4e4da8c0/pootle/apps/virtualfolder/models.py#L134-L138 obsolete units might get retrieved when calculating stats.

Have in mind that when relating units with a vfolder (either with `add_vfolders` or when relating new units using https://github.com/translate/pootle/blob/4e4da8c0/pootle/apps/virtualfolder/models.py#L359-L381), these are never obsolete, but they might get obsoleted afterwards.

The best way might be to remove the unit from all vfolders when it is obsoleted, and relate it to any existing vfolder when it is resurrected by using https://github.com/translate/pootle/blob/4e4da8c0/pootle/apps/virtualfolder/models.py#L359-L381.

**Comment by unho** [comment]
This will no longer apply once #3816 is merged. But perhaps still is a good idea to remove the unit from all vfolders when it is obsoleted, and relate it to any existing vfolder when it is resurrected.

**Comment by ta2-1** [comment]
@unho, will it be displayed empty Virtual folders with zero stats after merge #3816 ? We know about this issue and leave it for future, do we? Also do you think there can be any issues if a whole store or directory is marked as obsolete after sync?

**Comment by unho** [comment]
After #3816 vfolders with no calculated stats will be displayed if:

- the visitor is an admin or
- if the vfolder is public and has priority >= 1

If having no stats is considered a reason for not displaying the vfolder (I don't fully agree with this) then we can create a separate issue for it.

About the obsolete. I haven't tested. As far as I can tell mark a store as obsolete doesn't have any effect on stats for its directory (https://github.com/translate/pootle/blob/master/pootle/apps/pootle_store/models.py#L966-L973) and marking a directory as obsolete does not flush the stats for related directories (https://github.com/translate/pootle/blob/master/pootle/apps/pootle_app/models/directory.py#L277) so the effect on vfolders should be exactly the same.
